builtin/game/chatcommands.lua

```diff
@@ -451,59 +457,123 @@ core.register_chatcommand("set", {
 	end,
 })
 
-local function emergeblocks_callback(pos, action, num_calls_remaining, ctx)
-	if ctx.total_blocks == 0 then
-		ctx.total_blocks   = num_calls_remaining + 1
-		ctx.current_blocks = 0
+local function emerge_callback(pos, action, num_calls_remaining, ctx)
+	if ctx.total_count == 0 then
+		ctx.total_count   = num_calls_remaining + 1
+		ctx.current_count = 0
 	end
-	ctx.current_blocks = ctx.current_blocks + 1
-
-	if ctx.current_blocks == ctx.total_blocks then
+	ctx.current_count = ctx.current_count + 1
+	if os.clock() < ctx.cur_time then
+		-- os.clock() loops after 2^32 microseconds.
```

Comment:
If you address overflow:
`os.clock` is [cpu time in seconds since the programm was started](http://www.lua.org/manual/5.1/manual.html#pdf-os.clock)
with ms precision *on some platforms*

Since Lua uses 64bit doubles for everything, it should be able to store a few million years.
Or do they loop regardless? :fearful: 

Maybe consider using `os.time` with `os.difftime`. It's actual seconds, not cpu time used over all threads.